### PR TITLE
Callback for `PuoriborEnv.step`

### LINKED ISSUE #10: Standardize serialization format
Game serialization is not currently standardized:

https://github.com/poapper-inc/fights/blob/37392dbb5fc8f80146c714f38ebe52a3186fb23d/examples/puoribor.py#L63-L74

Currently, serialization is performed as: `(state, agent_id, action)` ➡️ `List[Dict[...]]` ➡️ MessagePack binary.

The `(state, agent_id, action)` ➡️ `List[Dict[...]]` step should be included in the `fights` package proper.

### PR DESCRIPTION
**Pull request by sohnryang**
This adds a callback support for `PuoriborEnv.step`.

### CODE REVIEW

File: fights/envs/puoribor.py [hunk 1]
```diff
@@ -152,7 +152,16 @@ class PuoriborEnv(BaseEnv):
...
+        self,
+        state: PuoriborState,
+        agent_id: int,
+        action: ArrayLike,
+        pre_callback: Union[
+            Callable[[PuoriborState, int, ArrayLike], None], None
+        ] = None,
+        post_callback: Union[
+            Callable[[PuoriborState, int, ArrayLike], None], None
+        ] = None,
```
Comment (yklcs) [inline fights/envs/puoribor.py:164]: `Optional[...]` should be preferred over `Union[..., None]`.
Also, let's make `pre_callback` and `post_callback` [keyword only arguments](https://peps.python.org/pep-3102/).

**Review by yklcs — CHANGES_REQUESTED** [review]

File: fights/envs/puoribor.py (hunk 1 above)
Comment (yklcs) [reply fights/envs/puoribor.py]: Is it just me, or does `pre_callback` sound a bit like a function that runs after `callback`?
Maybe we should call it `pre_step_fn` or  `pre_step_callback`  or `callback_pre` or something similar?

**Comment by yklcs** [comment]
Is it just me, or does `pre_callback` sound a bit like a function that runs after `callback`?
Maybe we should call it `pre_step_fn` or  `pre_step_callback`  or `callback_pre` or something similar?

**Comment by sohnryang** [comment]
In that regard, `callback_pre` also seems unclear to me, and `pre_step_callback` is a bit long. Use `pre_step_fn` then?

**Comment by yklcs** [comment]
`pre_step_fn` and `post_step_fn` sounds like the best option to me too.

**Review by yklcs — APPROVED** [review]
LGTM!

**Pull request merged**